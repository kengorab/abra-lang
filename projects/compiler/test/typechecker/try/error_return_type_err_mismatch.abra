func foo(): Result<Int, String> = Ok(123)

func bar(): Result<Int, Int> {
  val x = try foo()
  Ok(x + 1)
}