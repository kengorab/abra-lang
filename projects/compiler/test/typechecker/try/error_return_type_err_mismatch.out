Error at %FILE_NAME%:4:11
Return type mismatch for function containing try expression
  |    val x = try foo()
               ^
The containing function 'bar' returns a Result whose error type is 'Int', which is incompatible with the try expression.
  |  func bar(): Result<Int, Int> {
          ^
To be compatible, the return type of 'bar' must have 'String' as its error type.
Hint: You can use an else-clause to transform the error value into a compatible value and return it
